pgmpy/sampling/Sampling.py

```diff
@@ -1,15 +1,21 @@
-import itertools
 from collections import namedtuple
+import itertools
+from warnings import warn
 
 import networkx as nx
 import numpy as np
-from pandas import DataFrame
-from pgmpy.extern.six.moves import map, range
+try:
+    from pandas import DataFrame
+    HAS_PANDAS = True
```

Comment:
@khalibartan Rather than checking this in every file, we should move this to maybe `pgmpy/__init__.py` or some other separate file (along with other such dependency checks). And then we should import that variable in these files and use it. This will be much cleaner and move all the dependencies checking code in one place.